### ISSUE TITLE
Add Deep Learning CNN Model for Tomato Leaf Disease Prediction

### ISSUE THREAD

**Issue by keshripritesh** [omroy07/AgriTech#316]
### **📌 Issue Title:**

\[GSSOC’25] Add Deep Learning CNN Model for Tomato Leaf Disease Prediction

---

#### 🚀 Project Idea

This project proposes a **Deep Learning-based Convolutional Neural Network (CNN) model** that can predict various tomato plant diseases using leaf images. Early detection of plant diseases can help farmers take preventive measures and improve crop yield.

The model is trained on a **Tomato Leaf Image Dataset** and is capable of classifying tomato leaves into multiple disease categories, as well as detecting healthy leaves.

#### 🖼️ Diseases the Model Can Predict

* Bacterial spot
* Early blight
* Late blight
* Leaf mold
* Septoria leaf spot
* Spider mites (Two-spotted spider mite)
* Target spot
* Tomato Yellow Leaf Curl Virus
* **Healthy Tomato Leaf**

#### 📊 Dataset & Preprocessing

* Dataset: Tomato Leaf Image Dataset
* Preprocessing: Resizing all images to **(50, 50, 3)** for uniformity
* Augmentation applied for better generalization

#### 🏗️ Model Architecture

* A CNN-based classifier with multiple convolutional, pooling, and dense layers
* Final softmax layer for multiclass classification
* Achieved **Train Accuracy: 99.96%** and **Test Accuracy: 96.99%**

#### ✅ Key Features

* High accuracy with minimal overfitting
* Ability to classify **8 tomato diseases + healthy leaves**
* Potential to extend into a **web/mobile app** for real-world farmers

---

#### `Healthy Tomato Leaf`

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/eca1d28f-e39e-4a4c-8f2a-21bb7b7dd00b" />

#### `Tomato Leaf with Bacterial Spot`

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/a8d1773f-fe42-46c1-a32c-1878bd09b7d2" />

#### `Tomato Leaf with Septoria Leaf Spot`

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/c01f6c49-86bc-4bb2-94a5-6bbab69464ce" />

#### `Tomato Leaf with Yellow Leaf Curl Virus`

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/d9f844a8-014e-4a40-8f92-c95f56ff00cc" />

#### `Model Architecture`

<img width="683" height="957" alt="Image" src="https://github.com/user-attachments/assets/54258ec0-a9e9-42a5-b16e-0398880c67b7" />

#### `Model Accuracy`

<img width="1304" height="506" alt="Image" src="https://github.com/user-attachments/assets/7730c0a2-da95-4548-b509-b3c305401703" />

---

### **📍 Expected Outcome**

A **deep learning-based plant disease prediction system** that can:

* Classify tomato plant diseases from leaf images with **\~97% test accuracy**
* Provide a deployable solution for **real-world agricultural applications**
* Help farmers and researchers in **early detection & crop protection**

---
